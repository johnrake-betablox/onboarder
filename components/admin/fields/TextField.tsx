import { DMMF } from "@prisma/client/runtime";
import { ChangeEvent } from "react";
import FieldLabel from "../FieldLabel";

interface Props {
  field: DMMF.Field;
  value: string;
  onChange: (key: string, value: string) => void;
}
export default function TextField({ field, value, onChange }: Props) {
  const handleChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
    onChange(field.name, e.currentTarget.value);
  };

  return (
    <div>
      <FieldLabel field={field} />
      <textarea
        className="focus:shadow-outline w-full appearance-none rounded border px-3 py-2 leading-tight text-gray-700 shadow"
        id={field.name}
        name={field.name}
        rows={5}
        value={value || ""}
        required={field.isRequired}
        onChange={handleChange}
      />
    </div>
  );
}
