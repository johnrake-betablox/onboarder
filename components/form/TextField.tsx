interface Props {
  field: string;
  value: any;
  required?: boolean;
  type?: string;
  autoComplete?: string;
  onChange: (value: string) => void;
}
export default function TextField({
  field,
  value,
  onChange,
  type = "text",
  autoComplete,
  required = false,
}: Props) {
  return (
    <input
      id={field}
      name={field}
      type={type}
      required={required}
      autoComplete={autoComplete}
      className="mt-1 block w-full appearance-none rounded-md border border-gray-300 px-3 py-2 placeholder-gray-400 shadow-sm focus:border-green-600 focus:outline-none focus:ring-green-600 sm:text-sm"
      value={value}
      onChange={(e) => onChange(e.currentTarget.value)}
    />
  );
}
