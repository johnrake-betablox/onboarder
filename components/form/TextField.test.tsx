import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import TextField from "./TextField";

const onChange = (value: any) => {};

test("renders a snapshot", () => {
  const { asFragment } = render(
    <TextField field="firstName" value="John" onChange={onChange} />,
  );
  expect(asFragment()).toMatchSnapshot();
});

test("renders a required field", () => {
  const { asFragment } = render(
    <TextField field="firstName" value="John" onChange={onChange} required />,
  );
  expect(asFragment()).toMatchSnapshot();
});

test("renders with a specific type", () => {
  const { asFragment } = render(
    <TextField
      field="password"
      value="secret"
      onChange={onChange}
      type="password"
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});

test("renders with an autocomplete", () => {
  const { asFragment } = render(
    <TextField
      field="email"
      value="john@betablox.com"
      onChange={onChange}
      type="email"
      autoComplete="email"
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
