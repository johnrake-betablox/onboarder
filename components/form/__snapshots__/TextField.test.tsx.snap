// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a required field 1`] = `
<DocumentFragment>
  <input
    class="mt-1 block w-full appearance-none rounded-md border border-gray-300 px-3 py-2 placeholder-gray-400 shadow-sm focus:border-green-600 focus:outline-none focus:ring-green-600 sm:text-sm"
    id="firstName"
    name="firstName"
    required=""
    type="text"
    value="John"
  />
</DocumentFragment>
`;

exports[`renders a snapshot 1`] = `
<DocumentFragment>
  <input
    class="mt-1 block w-full appearance-none rounded-md border border-gray-300 px-3 py-2 placeholder-gray-400 shadow-sm focus:border-green-600 focus:outline-none focus:ring-green-600 sm:text-sm"
    id="firstName"
    name="firstName"
    type="text"
    value="John"
  />
</DocumentFragment>
`;

exports[`renders with a specific type 1`] = `
<DocumentFragment>
  <input
    class="mt-1 block w-full appearance-none rounded-md border border-gray-300 px-3 py-2 placeholder-gray-400 shadow-sm focus:border-green-600 focus:outline-none focus:ring-green-600 sm:text-sm"
    id="password"
    name="password"
    type="password"
    value="secret"
  />
</DocumentFragment>
`;

exports[`renders with an autocomplete 1`] = `
<DocumentFragment>
  <input
    autocomplete="email"
    class="mt-1 block w-full appearance-none rounded-md border border-gray-300 px-3 py-2 placeholder-gray-400 shadow-sm focus:border-green-600 focus:outline-none focus:ring-green-600 sm:text-sm"
    id="email"
    name="email"
    type="email"
    value="john@betablox.com"
  />
</DocumentFragment>
`;
